synfig-studio/src/gui/app.cpp

```diff
@@ -2332,36 +2332,32 @@ App::apply_gtk_settings()
 //	data += "button > image { padding: 5px; }\n";
 //	data += "combobox > box > button > box { padding-top: 0px; padding-bottom: 0px; }\n";
 
-	data += "#render-progress * { min-height: 20px; }\n";
-
-	data += ".synfigstudio-efficient-workspace button { padding: 0px; }\n";
-	data += ".synfigstudio-efficient-workspace button > image { padding: 5px; }\n";
-
-	data += "#low-resolution label { padding-left: 5px; padding-right: 5px; }\n";
-
-	data += ".synfigstudio-efficient-workspace entry, spinbutton { min-height: 16px; }\n";
+//#if __APPLE__
+//	// Fix #810: Insensitive context menus on OSX
+//	g_object_get (G_OBJECT (gtk_settings), "gtk-theme-name", &theme_name, NULL);
+//	if ( String(theme_name) == "Adwaita" )
+//		data += ".window-frame, .window-frame:backdrop { box-shadow: none; margin: 0; }\n";
+//	g_free(theme_name);
+//#endif
 
-	// We don't want the combobox icons touching the left/right edge. Give some lil' space.
-	data += ".synfigstudio-efficient-workspace combobox button { padding-left: 5px; padding-right: 5px; }\n";
+	auto provider = Gtk::CssProvider::create();
+	auto screen   = Gdk::Screen::get_default();
 
+	try {
 #if __APPLE__
-	// Fix #810: Insensitive context menus on OSX
-	g_object_get (G_OBJECT (gtk_settings), "gtk-theme-name", &theme_name, NULL);
-	if ( String(theme_name) == "Adwaita" )
-		data += ".window-frame, .window-frame:backdrop { box-shadow: none; margin: 0; }\n";
-	g_free(theme_name);
+		g_object_get (G_OBJECT (gtk_settings), "gtk-theme-name", &theme_name, NULL);
+		if ( String(theme_name) == "Adwaita" )
+			auto css_file = Gio::File::create_for_path(ResourceHelper::get_css_path("synfig.mac.css"));
+		g_free(theme_name);
```

Comment:
Maybe it's because of those lines

My guess is Adwaita condition check is getting skipped resulting in css_file being undeclared

Should I remove the theme code?

Reply:
The problem is that `css_file` gets destroyed when it goes out of scope from `if` condition. So you need to declare it before `if`.

Reply:
So .mac.css would be called only if theme is Adwaita?

Reply:
> So .mac.css would be called only if theme is Adwaita?

Yes. But I don't have a Mac with me so someone should test it on other themes and not only on Adwaita.

@ice0 Thanks for the scope :) Me personally I don't want to keep that if guard because the toolbar buttons might not be the same for other themes. And Apple users might complain synfig UI has become too thick again when other themes are applied.

@morevnaproject wrote that code I think. I read his post on [gtk mailing lists](http://gtk.10911.n7.nabble.com/Gtk3-MacOS-OSX-context-menu-issues-td86576.html)

My opinion... test if OS X has insensitive context menus issue for other GTK themes. If so then remove the if part and let OS X apply the generic theme like all other platforms.

Reply:
I agree. As temporary solution, I suggest to leave this check as-is, so we can merge your changes, and add an issue to check and fix that problem.

For out-of-scope variable. You can make a reorder, to fix this:
```
auto css_file = Gio::File::create_for_path(ResourceHelper::get_css_path("synfig.css"));

#ifdef __APPLE__
		g_object_get (G_OBJECT (gtk_settings), "gtk-theme-name", &theme_name, NULL);
		if ( String(theme_name) == "Adwaita" )
			css_file = Gio::File::create_for_path(ResourceHelper::get_css_path("synfig.mac.css"));
		g_free(theme_name);
#endif
```

Reply:
Won't the variable `css_file` names conflict in case Adwaita check turns out correct?
Something like this compiler error
> Variable already declared here...

Reply:
@ice0 What about like this?
 ```
#ifdef __APPLE__
		Glib::RefPtr<Gio::File css_file;
		g_object_get (G_OBJECT (gtk_settings), "gtk-theme-name", &theme_name, NULL);
		if ( String(theme_name) == "Adwaita" )
			css_file = Gio::File::create_for_path(ResourceHelper::get_css_path("synfig.mac.css"));
		g_free(theme_name);
#endif
```

Reply:
Proceeding with your suggestion anyway :) Mine's still flawed in some cases...

Reply:
> Won't the variable css_file names conflict in case Adwaita check turns out correct?

No, it just replace value. As you can notice I removed the ʻauto` keyword.

Reply:
Noticed OS X build succeeded on Travis :)

Now, pending tasks left are:
- padding timeslider
- removing commented code after final review's done